File: Programming EF Code First/chapter4/Chapter4/BAGA_FluentAPI/DataAccess/InternetSpecialConfiguration.cs
using System.Data.Entity.ModelConfiguration;
using Model;
using System.ComponentModel.DataAnnotations;

namespace DataAccess
{
  public class InternetSpecialConfiguration : EntityTypeConfiguration<InternetSpecial>
  {
    public InternetSpecialConfiguration()
    {
      HasRequired(s => s.Accommodation)
        .WithMany(l => l.InternetSpecials)
        .HasForeignKey(s => s.AccommodationId);
    }
  }
}

File: Programming EF Code First/chapter5/Chapter 5 VB/Fluent/DataAccess/InternetSpecialConfiguration.vb
Imports System.Data.Entity.ModelConfiguration
Imports Model
Imports System.ComponentModel.DataAnnotations

Namespace DataAccess
  Public Class InternetSpecialConfiguration
	  Inherits EntityTypeConfiguration(Of InternetSpecial)
	Public Sub New()
	  HasRequired(Function(s) s.Accommodation).WithMany(Function(l) l.InternetSpecials).HasForeignKey(Function(s) s.AccommodationId)
	End Sub
  End Class
End Namespace